endpoints/citrus-kafka/src/test/java/org/citrusframework/kafka/endpoint/KafkaConsumerSelectorTest.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citrusframework.kafka.endpoint;
+
+import org.citrusframework.exceptions.CitrusRuntimeException;
+import org.citrusframework.kafka.endpoint.matcher.KafkaConsumerRecordMatcher;
+import org.citrusframework.kafka.endpoint.matcher.KafkaConsumerRecordMatcherFactory;
+import org.citrusframework.message.MessageSelectorBuilder;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.citrusframework.kafka.endpoint.KafkaConsumerSelector.EVENT_LOOKBACK_WINDOW;
+import static org.citrusframework.kafka.endpoint.KafkaConsumerSelector.POLL_TIMEOUT;
+import static org.citrusframework.kafka.endpoint.KafkaConsumerSelector.kafkaConsumerSelector;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class KafkaConsumerSelectorTest {
+
+    private KafkaConsumerSelector fixture;
+
+    @BeforeMethod
+    public void beforeMethodSetup() {
+        fixture = kafkaConsumerSelector().buildSelector();
+    }
+
+    @Test
+    public void hasKafkaConsumerRecordMatcherFactory() {
+        assertThat(fixture.getKafkaConsumerRecordMatcherFactory())
+                .isNotNull();
+    }
+
+    @Test
+    public void defaultPollTimeout() {
+        assertThat(fixture.getPollTimeout())
+                .isEqualTo(Duration.ofMillis(100));
```

Comment:
Seems to me like a very short time for default poll period. But i might be wrong.

Reply:
it's the time it takes to read one message, not the general poll timeout. I've even thought about just setting this statically. no clue if it's reasonable, but from what I've read it seems to be.